sonar-scanner-engine/src/main/java/org/sonar/scanner/repository/settings/SettingsLoader.java

```diff
@@ -21,6 +21,7 @@
 
 import java.util.Map;
 
+@FunctionalInterface
 public interface SettingsLoader {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Drop this interface in favor of "java.util.function.Function<String,Map>". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1711)